m4opt/tests/cerenkov_test/test_cerenkov.py

```diff
@@ -0,0 +1,400 @@
+            [6, 107.712056, 20.3254414, 91.3718367, 493.910694],
+            [6.5, 77.8067567, 13.5646922, 63.4974925, 358.767662],
+            [7, 57.4414259, 9.37080518, 45.3675733, 265.858268],
+            [8.5, 21.5238149, 3.0546999, 15.8521726, 100.141456],
+        ]
+    )
+
+    arr = np.column_stack(
+        [
+            Fmat1,
+            Fmat2[:, 2:4],
+            Fmat3[:, 2:4],
+        ]
+    )
+    colnames = [
+        "Energy",
+        "DailyMin_MeanFlux",
+        "DailyMax_MeanFlux",
+        "DailyMin_95Flux",
+        "DailyMax_95Flux",
+        "DailyMax_50Flux",
+        "DailyMax_75Flux",
+        "DailyMin_50Flux",
+        "DailyMin_75Flux",
+    ]
+    return Table(arr, names=colnames)
+
+
+def cerenkov_emission_mat(xml_path, fields=None):
+    r"""
+    Reads a MATLAB struct-style XML file and returns the results as an Astropy Table.
+
+    The XML file contains the Cerenkov emission calculated using the daily flux (DailyMax)
+    at the 95th percentile, for the material: 'si02_suprasil_2a'.
+
+    Args:
+        xml_path (str): Path to the XML file.
+        fields (list or None): List of fields/tags to extract. If None, detects all automatically.
+
+    Returns:
+        astropy.table.Table : Table containing the results as columns.
+
+        - 'Lam' : np.ndarray
+            Wavelength array [Å] (Angström, equivalent to Ang or \(\AA\)).
+        - 'Int' : np.ndarray
+            Cerenkov intensity [count/cm^2/s/sr/μm].
+        - 'Int_Units' : str
+            Units of intensity, 'count/cm^2/s/sr/μm'.
+        - 'IntAA' : np.ndarray
+            Intensity per arcsecond squared [count/cm^2/s/arcsec^2/Ang].
+        - 'IntAA_Units' : str
+            Units for 'IntAA', 'count/cm^2/s/arcsec^2/Ang'.
+        - 'IntFA' : np.ndarray
+            Energy flux in erg/cm^2/s/arcsec^2/Ang.
+        - 'IntFA_Units' : str
+            Units for 'IntFA', 'erg/cm^2/s/arcsec^2/Ang'.
+        - 'n' : np.ndarray
+            Refractive index values for each wavelength.
+        - 'Lum' : np.ndarray
+            Luminosity per wavelength [count/cm^2/s/μm].
+        - 'Int_arcsec_Units' : str
+            Optional label for display: 'counts cm$^{-2}$ s$^{-1}$ arcsec$^{-2}$ \AA$^{-1}$'.
+    """
+
+    tree = ET.parse(xml_path)
+    root = tree.getroot()
+
+    # Auto-detect fields
+    if fields is None:
+        fields = list({child.tag for child in root if not child.tag.endswith("_Units")})
+
+    # Extract values for each field
+    data = {}
+    for tag in fields:
+        try:
+            data[tag] = [float(e.text) for e in root.findall(tag)]
+        except Exception:
+            data[tag] = [e.text for e in root.findall(tag)]
+
+    # Truncate to minimum length (in case of unequal arrays)
+    min_len = min(len(lst) for lst in data.values())
+    data = {k: v[:min_len] for k, v in data.items()}
+
+    # Create and return Astropy Table
+    return Table(data)
+
+
+def cerenkov_emission_py(
```

Comment:
A lot of this looks like it is copy-pasted from your code in `m4opt.synphot.background._cerenkov`. Why? It's not a valid test if you are merely testing the code against a copy of itself.